feat(backend): run wallet grpc server with a password

alex1307 opened the pull request:
Description

We will run the grpc server using a session password: the password is provided when the wallet is started.
Or the password could be configred using the conf. variable: TARI_WALLET_GRPC_PASSWORD.

Motivation and Context
We should protect the wallet sensitive infromation.

How Has This Been Tested?
Both cases are maunally tested.
Run the wallet with -- --grpc-password=12345678A and called getBalance.
When the correct password is provide balance is shown.
If not then 16 UNAUTHENTICATED: Invalid gRpc password is shown.

Run without a password and balance is shown with or without authorization.

Setup:
grpc-password=12345
Metadata: {"authorization": "7d4e3eec80026719639ed4dba68916eb94c7a49a053e05c8f9578fe4e5a3d7ea"}

alex1307 committed:
feat(backend): run wallet grpc server with a password.
Description

We will run the grpc server using a session password: the password is provided when the wallet is started.
Or the password could be configred using the conf. variable: TARI_WALLET_GRPC_PASSWORD.

Motivation and Context
We should protect the wallet sensitive infromation.

How Has This Been Tested?
Both cases are maunally tested.
Run the wallet with -- --grpc-password=12345678A and called getBalance.
When the correct password is provide balance is shown.
If not then 16 UNAUTHENTICATED: Invalid gRpc password is shown.

Run without a password and balance is shown with or without authorization.

CjS77 commented on applications/tari_console_wallet/src/main.rs:
```diff
@@ -101,6 +101,11 @@ fn main_inner() -> Result<(), ExitError> {
+        println!("GRPC password: {}", grpc_password);
+    } else {
+        println!("GRPC password is NOT FOUND");
+    }
```
Don't print sensitive information like passwords.

CjS77 reviewed (commented):
I don't know that this is secure enough. We should at least be hashing passwords and using other best practices.

alex1307 committed:
Hashing the grpc wallet server password

alex1307 committed:
feat(backend): restart docker containers
Description

Applying formattng.
Removing useBootstrapper field and fixing compilation bug.
Bypassing base_layer/p2p/src/transport.rs lints clippy check.
Fixing compilation issue.

Motivation and Context
Clean code quality requirement.

How Has This Been Tested?
The change is tested manually.

alex1307 committed:
Merge branch 'launchpad_such_wow' into 268_wallet_grpc_password

CjS77 commented:
I've been thinking about this. Let's rather do this properly with TLS.

Since that's a slightly bigger effort, let's deprioritise it for now.

CjS77 commented on applications/launchpad/backend/src/docker/mod.rs:
```diff
@@ -151,3 +152,27 @@ pub async fn try_destroy_container(image_name: &str, docker: Docker) -> Result<(
     Ok(())
 }
+
+pub fn random_password() -> String {
+    use rand::Rng;
+    const CHARSET: &[u8] = b"ABCDEFGHIJKLMNOPQRSTUVWXYZ\
+                            abcdefghijklmnopqrstuvwxyz\
+                            0123456789)(*&^%$#@!~";
+    const PASSWORD_LEN: usize = 30;
+    let mut rng = rand::thread_rng();
+
+    (0..PASSWORD_LEN)
+        .map(|_| {
+            let idx = rng.gen_range(0..CHARSET.len());
+            CHARSET[idx] as char
+        })
+        .collect()
+}
```
There's already a password generator function in `helpers` btw

CjS77 commented:
Going to shelve this approach for a more robust TLS-based one
